src/Utilities/Compiler/Options/SymbolNamesOption.cs

```diff
@@ -41,14 +53,58 @@ private SymbolNamesOption()
 
             var namesBuilder = PooledDictionary<string, string?>.GetInstance();
             var symbolsBuilder = PooledDictionary<ISymbol, string?>.GetInstance();
+            var wildcardNamesBuilder = PooledDictionary<SymbolKind, PooledDictionary<string, string?>>.GetInstance();
 
             foreach (var symbolName in symbolNames)
             {
                 var parts = getSymbolNamePartsFunc != null
                     ? getSymbolNamePartsFunc(symbolName)
                     : new NameParts(symbolName);
 
-                if (parts.SymbolName.Equals(".ctor", StringComparison.Ordinal) ||
+                var numberOfWildcards = symbolName.Count(c => c == WildcardChar);
```

Comment:
Nit: this method seems to have become big enough to warrant breaking it into different local functions for each bucket (symbols, names, wildcard based names)